Editor/EditorCore/EditorMeshHandles.cs

```diff
@@ -238,35 +238,35 @@ public static void DrawSceneSelection(SceneSelection selection)
             var positions = mesh.positionsInternal;
 
             // Draw nearest edge
-            if (selection.face != null)
+            using (new TriangleDrawingScope(s_PreselectionColor))
             {
-                using (new TriangleDrawingScope(s_PreselectionColor))
+                foreach (var face in selection.faces)
                 {
                     GL.MultMatrix(mesh.transform.localToWorldMatrix);
```

Comment:
Same note for `GL.MultMatrix`, this only needs to be done once per-object, then all faces should use that same matrix.